owncloudApp/src/main/java/com/owncloud/android/MainApp.kt

```diff
@@ -207,8 +214,28 @@ class MainApp : Application() {
         )
     }
 
+    private fun handleRestrictions(activity: Activity) {
+        val restrictionsManager = activity.getSystemService(Context.RESTRICTIONS_SERVICE) as RestrictionsManager
+        val restrictions = restrictionsManager.applicationRestrictions
+        cacheRestrictions(activity, restrictions)
+    }
+
+    private fun cacheRestrictions(activity: Activity, restrictions: Bundle) {
+        val preferencesProvider = SharedPreferencesProviderImpl(activity)
+        if (restrictions.containsKey(CONFIGURATION_SERVER_URL)) {
+            val managedString = restrictions.getString(CONFIGURATION_SERVER_URL)
+            managedString?.let { preferencesProvider.putString(CONFIGURATION_SERVER_URL, it) }
+        }
+        if (restrictions.containsKey(CONFIGURATION_SERVER_URL_INPUT_VISIBILITY)) {
+            val managedBool = restrictions.getBoolean(CONFIGURATION_SERVER_URL_INPUT_VISIBILITY)
+            preferencesProvider.putBoolean(CONFIGURATION_SERVER_URL_INPUT_VISIBILITY, managedBool)
+        }
+    }
+
     companion object {
         private const val BETA_VERSION = "beta"
+        const val CONFIGURATION_SERVER_URL = "server_url_configuration"
+        const val CONFIGURATION_SERVER_URL_INPUT_VISIBILITY = "server_url_input_visibility_configuration"
```

Comment:
Now we have 2 constants and it could seem a good place to add those constants, but how will this companion object look like after 30 configurations?

Let's create a new file, maybe `MdmConfigurations.kt`, and add there those constants. It will be easier to maintain and MainApp will be cleaner 👍